server/src/uuid/model/entity/mod.rs

```diff
@@ -410,8 +410,9 @@ impl Entity {
                 SELECT c.id
                     FROM entity_link l
                     JOIN entity c on c.id = l.child_id
+                    JOIN uuid on uuid.id = c.id
                     JOIN type t ON t.id = c.type_id
-                    WHERE l.parent_id = ? AND t.name = ?
+                    WHERE l.parent_id = ? AND t.name = ? AND uuid.trashed = 0
```

Comment:
```suggestion
                    WHERE l.parent_id = ? AND t.name = ?
```
Since `fild_childen_by_id_and_type` are used by other cases, not just solution, we should keep it as it is.

I know it was in the issue, sorry for formulating there in a very misleading way. 
What I had in mind was to filter out trashed in `fild_childen_by_id_and_type` in `fild_child_by_id_and_type`, something like:
```rust
    async fn find_child_by_id_and_type (...) ...
    {
        Self::find_children_by_id_and_type(id, child_type, executor)
            .await
            .filter(|children| children.trashed == 0)
            .map(|children| children.first().cloned())
    }
```
For that you need of course to give this information in line 410 
```sql
SELECT c.id, trashed
```

Reply:
Returning the trashed info might not result in the nicest code but I'll try and see.